### PR TITLE
[API] Get scopes of the adherent

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/Api/ScopeController.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+use Symfony\Component\Routing\Annotation\Route;
+
+class ScopeController extends AbstractController
+{
+    /**
+     * @Route("/v3/scopes", name="api_scopes", methods={"GET"})
+     */
+    public function __invoke(GeneralScopeGenerator $scopeGenerator): JsonResponse
+    {
+        if (!$this->getUser()) {
```
Comment (ottaviano) [inline src/Controller/Api/ScopeController.php:19]: Imo, we can avoid this IF and inject `UserInterface $user` directly

File: src/Scope/Generator/DeputyScopeGenerator.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+use App\Entity\Adherent;
+use App\Scope\AppEnum;
+use App\Scope\ScopeEnum;
+use App\Scope\ScopeValueObject;
+
+class DeputyScopeGenerator implements ScopeGeneratorInterface
+{
+    public function generate(Adherent $adherent): ScopeValueObject
+    {
+        $scope = new ScopeValueObject(
```
Comment (ottaviano) [inline src/Scope/Generator/DeputyScopeGenerator.php:14]: `$scope` is useless var

File: src/Scope/Generator/GeneralScopeGenerator.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Scope\Generator;
```
Comment (ottaviano) [inline src/Scope/Generator/GeneralScopeGenerator.php:3]: what about moving this general generator to App\Scope namespace?

File: src/Scope/ScopeValueObject.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Scope;
+
+use App\Entity\Geo\Zone;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Symfony\Component\Serializer\Annotation as SymfonySerializer;
+
+class ScopeValueObject
```
Comment (ottaviano) [inline src/Scope/ScopeValueObject.php:10]:
```suggestion
class Scope
```

File: src/Scope/ScopeValueObject.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+     * @var array
+     *
+     * @SymfonySerializer\Groups({"scopes"})
+     */
+    private $apps;
+
+    public function __construct(string $code, array $zones, array $apps)
+    {
+        $this->code = $code;
+        $this->zones = new ArrayCollection($zones);
```
Comment (ottaviano) [inline src/Scope/ScopeValueObject.php:36]: why do you use Doctrine class here?

Comment (illusionOfParadise) [reply src/Scope/ScopeValueObject.php]: No need, I removed it.

File: tests/Scope/GeneralScopeGeneratorTest.php [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        $this->generalScopeGenerator = $this->get(GeneralScopeGenerator::class);
+        $this->adherentRepository = $this->get(AdherentRepository::class);
+        $this->serializer = $this->get(SerializerInterface::class);
+    }
+
+    protected function tearDown(): void
+    {
+        $this->generalScopeGenerator = null;
+        $this->adherentRepository = null;
+        $this->serializer = null;
```
Comment (ottaviano) [inline tests/Scope/GeneralScopeGeneratorTest.php:108]: seems to be unused

**Review by ottaviano — APPROVED** [review]

**Pull request merged**